include/core/CProcessPriority.h

```diff
@@ -26,19 +26,21 @@ namespace core {
 //! IMPLEMENTATION DECISIONS:\n
 //! This is a static class - it's not possible to construct an instance of it.
 //!
-//! The basic implementation does nothing.  Platform-specific implementations
-//! exist for platforms where we have decided to do something.
+//! On Linux and macOS we reduce the CPU scheduling priority using "nice".
 //!
-//! Currently the only platform-specific implementation is for Linux, and it
-//! attempts to increase the OOM killer adjustment for the process such that
-//! it is more likely to be killed than other processes when the Linux kernel
-//! decides that there isn't enough free memory.
+//! On Linux we also attempt to increase the OOM killer adjustment for the
+//! process such that it is more likely to be killed than other processes
+//! when the Linux kernel decides that there isn't enough free memory.
 //!
 class CORE_EXPORT CProcessPriority : private CNonInstantiatable {
 public:
-    //! Reduce whatever priority measures are deemed appropriate for the
+    //! Reduce priority for memory if deemed appropriate for the
     //! current OS.
-    static void reducePriority();
+    static void reduceMemoryPriority();
+
+    //! Reduce priority for CPU if deemed appropriate for the
+    //! current OS.
+    static void reduceCpuPriority();
```

Comment:
How about a `static void reducePriority()` method which calls both as well? We generally always call both.

Reply:
The reason I didn't do this is because we want to reduce memory priority before installing system call filtering, but want to reduce CPU priority after connecting the named pipes.

Maybe that's premature optimisation, but the named pipe connection code is quite time-sensitive and you don't get any useful errors if it fails for an obscure reason.  I was worried that it might sporadically fail due to lack of CPU but this would be impossible to prove.

I can add this reasoning to the Doxygen comment of the class for future reference.

Reply:
Ok that makes sense. I think this is worth a comment at the point at which you call the `reduceCpuPriority`.